model/config.go

```diff
@@ -180,11 +180,11 @@ func (i *IstioRegistry) RouteRulesBySource(namespace string, instances []*Servic
 	rules := make([]*routeRuleConfig, 0)
 	for key, rule := range i.RouteRules(namespace) {
 		// validate that rule match predicate applies to source service instances
-		if rule.Match != nil {
+		if rule.Match != nil && rule.Match.Source != "" {
 			found := false
```

Comment:
@rshriram: this is just a simple coding bug. Any rule without a Match.Source field means "any source", so the function is supposed to return that rule, regardless of who's asking. But the bug is because Ingress calls this function with an empty instance list, and in that case it was doing the wrong thing with (i.e., discarding) the non-source-specific rules.